In [1]:
import pymc3 as pm

C:\Users\Arjun Janamatti\Anaconda3\envs\tf_nptel\lib\site-packages\numpy\core\__init__.py:29: UserWarning: loaded more than 1 DLL from .libs:
C:\Users\Arjun Janamatti\Anaconda3\envs\tf_nptel\lib\site-packages\numpy\.libs\libopenblas.PYQHXLVVQ7VESDPUVUADXEVJOBGHJPAY.gfortran-win_amd64.dll
C:\Users\Arjun Janamatti\Anaconda3\envs\tf_nptel\lib\site-packages\numpy\.libs\libopenblas.TXA6YQSD3GCQQC22GEQ54J2UDCXDXHWN.gfortran-win_amd64.dll
  stacklevel=1)
WARNING (theano.configdefaults): g++ not available, if using conda: `conda install m2w64-toolchain`
C:\Users\Arjun Janamatti\Anaconda3\envs\tf_nptel\lib\site-packages\theano\configdefaults.py:697: UserWarning: DeprecationWarning: there is no c++ compiler.This is deprecated and with Theano 0.11 a c++ compiler will be mandatory
  "DeprecationWarning: there is no c++ compiler."
WARNING (theano.configdefaults): g++ not detected ! Theano will be unable to execute optimized C-implementations (for both CPU and GPU) and will default to Python impleme

In [10]:
def get_plausible_values(observed):
    with pm.Model() as model_a:
        # Parameter = prior solution space
        conversion_rate = pm.Beta(name='conversion_rate_a',
                              alpha=2, 
                              beta=10)

        # Evaluator = likelihodd function
        how_likely = pm.Binomial(name = 'how_likely', 
                                 n = 100, 
                                 p = conversion_rate, 
                                 observed = observed)

        # Posterior solution space
        plausible_values_a = pm.sample()
    return plausible_values_a

In [ ]:
plausible_values_a = get_plausible_values(observed=7)
plausible_values_b = get_plausible_values(observed=17)

C:\Users\Arjun Janamatti\Anaconda3\envs\tf_nptel\lib\site-packages\pymc3\sampling.py:468: FutureWarning: In an upcoming release, pm.sample will return an `arviz.InferenceData` object instead of a `MultiTrace` by default. You can pass return_inferencedata=True or return_inferencedata=False to be safe and silence this warning.
  FutureWarning,
Auto-assigning NUTS sampler...
Initializing NUTS using jitter+adapt_diag...
Multiprocess sampling (4 chains in 4 jobs)
NUTS: [conversion_rate_a]


In [6]:
with pm.Model() as model_a:
    # Parameter = prior solution space
    conversion_rate = pm.Beta(name='conversion_rate_a',
                          alpha=2, 
                          beta=10)
    
    # Evaluator = likelihodd function
    how_likely = pm.Binomial(name = 'how_likely', 
                             n = 100, 
                             p = conversion_rate, 
                             observed = 7)
    
    # Posterior solution space
    plausible_values_a = pm.sample()

C:\Users\Arjun Janamatti\Anaconda3\envs\tf_nptel\lib\site-packages\pymc3\sampling.py:468: FutureWarning: In an upcoming release, pm.sample will return an `arviz.InferenceData` object instead of a `MultiTrace` by default. You can pass return_inferencedata=True or return_inferencedata=False to be safe and silence this warning.
  FutureWarning,
Auto-assigning NUTS sampler...
Initializing NUTS using jitter+adapt_diag...
Multiprocess sampling (4 chains in 4 jobs)
NUTS: [conversion_rate_a]


Sampling 4 chains for 1_000 tune and 1_000 draw iterations (4_000 + 4_000 draws total) took 31 seconds.
The acceptance probability does not match the target. It is 0.880973690623816, but should be close to 0.8. Try to increase the number of tuning steps.
The acceptance probability does not match the target. It is 0.8899293872194785, but should be close to 0.8. Try to increase the number of tuning steps.


In [7]:
# how many number of conversion rates are smaller than 0.1
plausible_values_a['conversion_rate_a'] < 0.1

array([ True,  True,  True, ...,  True,  True,  True])

In [8]:
np.mean(plausible_values_a['conversion_rate_a'] < 0.1)

0.7845

* 78.45% probability that the conversion rate is less than 10 %

In [ ]:
a_lt_b = plausible_values_a